dev-requirements.txt

```diff
@@ -2,6 +2,6 @@ build
 conda-lock
 pre-commit
 pytest-cov
-pytest>=4.6
+pytest>=7,<9
```

Comment:
preemptively blocking pytest 9? Are there known planned incompatibilities to deal with?

Reply:
Nothing known, I'm just fed up of dealing with random breakages! Do you want me to remove the upper limit?

Reply:
Ya, I think we should remove the limit as otherwise we may get stuck here. In general I think all upper limit pins should come with a time expiry.

Otherwise happy to merge this.